ca/certificate-authority-data.go

```diff
@@ -64,54 +64,23 @@ func (cadb *CertificateAuthorityDatabaseImpl) CreateTablesIfNotExists() (err err
 	return
 }
 
-// Begin starts a Database transaction. There can only be one in this object
-// at a time.
-func (cadb *CertificateAuthorityDatabaseImpl) Begin() (err error) {
-	if cadb.activeTx != nil {
-		err = errors.New("Transaction already open")
-		return
-	}
-	cadb.activeTx, err = cadb.dbMap.Begin()
-	return
-}
-
-// Commit makes permanent a database transaction; there must be an active
-// transaction when called.
-func (cadb *CertificateAuthorityDatabaseImpl) Commit() (err error) {
-	if cadb.activeTx == nil {
-		err = errors.New("Transaction already closed")
-		return
-	}
-	err = cadb.activeTx.Commit()
-	cadb.activeTx = nil
-	return
-}
-
-// Rollback cancels the ongoing database transaction; there must be an active
-// transaction when called.
-func (cadb *CertificateAuthorityDatabaseImpl) Rollback() (err error) {
-	if cadb.activeTx == nil {
-		err = errors.New("Transaction already closed")
-		return
-	}
-	err = cadb.activeTx.Rollback()
-	cadb.activeTx = nil
-	return
+// GetDbMap gets a pointer to the CA DB's GORP map object.
+func (cadb *CertificateAuthorityDatabaseImpl) GetDbMap() *gorp.DbMap {
```

Comment:
Actually, it looks like you are only using the DbMap to get a transaction.  So it would be safer just to expose a `cadb.Begin()` method that calls through to the `DbMap`.